src/com/xilinx/rapidwright/rwroute/RouteNodeGraph.java

```diff
@@ -175,69 +182,115 @@ protected RouteNodeGraph(Design design, RWRouteConfig config, Map<Tile, RouteNod
+                                } else {
+                                    assert(baseWireName.endsWith("_FT1"));
+                                    assert(baseTile.getTileYCoordinate() == intTile.getTileYCoordinate() + 1);
+                                }
                             }
+                            wires.set(baseNode.getWireIndex());
+                            continue;
                         }
-                        wires.set(baseNode.getWireIndex());
-                        continue;
-                    }
-                } else if (isUltraScale) {
-                    if (baseWireName.startsWith("INT_NODE_SINGLE_DOUBLE_") || baseWireName.startsWith("SDND") ||
-                            baseWireName.startsWith("INT_NODE_QUAD_LONG") || baseWireName.startsWith("QLND")) {
-                        if (baseTile != intTile) {
-                            if (baseWireName.endsWith("_FTN")) {
-                                assert (baseTile.getTileYCoordinate() == intTile.getTileYCoordinate() - 1);
-                            } else {
-                                assert (baseWireName.endsWith("_FTS"));
-                                assert (baseTile.getTileYCoordinate() == intTile.getTileYCoordinate() + 1);
+                    } else {
+                        assert(isUltraScale);
+                        if (baseWireName.startsWith("INT_NODE_SINGLE_DOUBLE_") || baseWireName.startsWith("SDND") ||
+                                baseWireName.startsWith("INT_NODE_QUAD_LONG") || baseWireName.startsWith("QLND")) {
+                            if (baseTile != intTile) {
+                                if (baseWireName.endsWith("_FTN")) {
+                                    assert(baseTile.getTileYCoordinate() == intTile.getTileYCoordinate() - 1);
+                                } else {
+                                    assert(baseWireName.endsWith("_FTS"));
+                                    assert(baseTile.getTileYCoordinate() == intTile.getTileYCoordinate() + 1);
+                                }
                             }
+                            wires.set(baseNode.getWireIndex());
+                            continue;
                         }
-                        wires.set(baseNode.getWireIndex());
-                        continue;
                     }
+                } else if (baseIntentCode != IntentCode.NODE_PINFEED && baseIntentCode != IntentCode.NODE_PINBOUNCE) {
+                    continue;
                 }
-            } else if (baseIntentCode != IntentCode.NODE_PINFEED && baseIntentCode != IntentCode.NODE_PINBOUNCE) {
-                continue;
-            }
+            } else {
+                assert(isVersal);
 
-            if (eastWestPattern == null) {
-                // FIXME:
-                assert(series == Series.Versal);
-                continue;
+                if (!EnumSet.of(IntentCode.NODE_IMUX, IntentCode.NODE_PINBOUNCE, IntentCode.NODE_INODE,
```

Comment:
Would it make sense to statically construct this set?

Reply:
I've hoisted it outside of the for loop.

Given that this set is only used inside the constructor, and the `RouteNodeGraph` object is only instantiated once per RWRoute invocation, I'd prefer the code readability benefits over making it a static object in the class body.